### PR TITLE
[#123] [Chore] Update Home Screen to be able to preview

### LINKED ISSUE #123: [Chore] Update Home Screen to be able to preview
## Why

Currently, `HomeScreen` is not able to preview. Make `HomeScreen` to be able to preview.

## Who Benefits?

Developer

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
Closes #123 

## What happened 👀

`HomeScreen` is not able to preview because of the viewModel.

## Insight 📝

- Decouple contents of `HomeScreen` from the main screen component
- Refactor `HomeViewModel` to be able to `Preview` different `UiState` of the `HomeScreen`

## Proof Of Work 📹

Screen Shot 2022-10-27 at 3 11 45 PM

### CODE REVIEW

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/home/HomeViewModel.kt [hunk 1]
```diff
@@ -1,75 +1,89 @@
...
-    private var _error = MutableStateFlow<ErrorModel?>(null)
-    private val _isRefreshing = MutableStateFlow(false)
-    private val _surveyList = MutableStateFlow<List<Survey>>(listOf())
-    private val _userAvatar = MutableStateFlow<String?>(null)
     private var _pageCount = DEFAULT_PAGES
     private var _records = DEFAULT_PAGES
-    private val _selectedSurveyNumber = MutableStateFlow(START_SURVEY_NUMBER)
+    private val _homeUiState = MutableStateFlow(HomeUiState())
+
+    val homeUiState: StateFlow<HomeUiState>
```
Comment (AVI5HEK) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/home/HomeViewModel.kt]: How about placing them together?
```suggestion
    private val _homeUiState = MutableStateFlow(HomeUiState())
    val homeUiState: StateFlow<HomeUiState>
```

**Review by AVI5HEK — COMMENTED** [review]
All in all, lgtm.

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/common/PreviewAnnotations.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    showBackground = true,
+    uiMode = UI_MODE_NIGHT_NO
+)
+@Preview(
+    name = "Night theme",
+    showSystemUi = true,
+    showBackground = true,
+    uiMode = UI_MODE_NIGHT_YES
+)
+annotation class DayNightPreviews
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/common/PreviewAnnotations.kt:20]: This is interesting, I like it 😄

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt [hunk 3]
```diff
@@ -288,28 +330,36 @@ fun BottomView(
...
+                },
+                onClick = onClickSurveyDetail
+            )
             Spacer(modifier = Modifier.width(20.dp))
         }
     }
 }
 
+@Composable
+fun SurveyRoundedButton(modifier: Modifier = Modifier, onClick: () -> Unit) {
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt]: Do we need this composable? Why not just use a `FloatingActionButton` instead? 🤔

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/home/HomeViewModel.kt [hunk 4]
```diff
@@ -1,75 +1,89 @@
...
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 const val DEFAULT_PAGE_SIZE = 5
 const val DEFAULT_PAGES = 1
 private const val START_SURVEY_NUMBER = 0
 
+data class HomeUiState(
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/home/HomeViewModel.kt]: This should probably be in its own file along with the `PreviewParameterProvider` below, what do you think? 🤔

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/home/HomeViewModel.kt [hunk 5]
```diff
@@ -1,75 +1,89 @@
...
+    val surveyList: List<Survey> = emptyList(),
+    val userAvatar: String? = null,
+    val error: ErrorModel? = null,
+    val isRefreshing: Boolean = false,
+    val selectedSurveyNumber: Int = START_SURVEY_NUMBER
+)
+
+class HomeUiStatePreviewParameterProvider : PreviewParameterProvider<HomeUiState> {
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/home/HomeViewModel.kt]: This is really cool, didn't know this existed 👏🏻

**Review by ryan-conway — COMMENTED** [review]
Just a few concerns, the rest LGTM 👍🏻

**Review by AVI5HEK — APPROVED** [review]
🚀

**Review by ryan-conway — APPROVED** [review]
🙏🏻

**Pull request merged**